src/App.tsx

```diff
@@ -15,71 +15,76 @@ export const goods = [
+    const { selectedGood } = this.state;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    this.setState({
+      selectedGood: selectedGood !== clickedGood ? clickedGood : undefined,
+    });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+    return (
+      <main className="section container">
+        {selectedGood ? (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedGood} is selected`}
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
+              className="delete ml-3"
+              onClick={() => this.setState({ selectedGood: undefined })}
             >
-              -
+              <span className="visually-hidden">Clear selected good</span>
             </button>
-          </td>
+          </h1>
+        ) : (
+          <h1 className="title">No goods selected</h1>
+        )}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map((good) => {
+              const isSelected = selectedGood === good;
+              const buttonClassName = `button ${
+                isSelected ? 'has-background-success-light' : ''
+              }`;
+              const minusStyle = isSelected ? { color: 'white' } : {};
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              return (
+                <tr key={good} data-cy="Good">
+                  <td>
+                    <button
+                      data-cy="AddButton"
+                      type="button"
+                      className={`${buttonClassName}`}
+                      onClick={() => this.handleButtonClick(good)}
+                      style={minusStyle}
+                    >
+                      {isSelected ? <span style={minusStyle}>-</span> : '+'}
+                    </button>
+                  </td>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+                  <td data-cy="GoodTitle" className="is-vcentered">
+                    {good}
+                  </td>
+                </tr>
```

Comment:
Please, double check your styles. At that moment I still see white background on active item:
![image](https://github.com/mate-academy/react_goods-selector/assets/42767822/e7b0cf4f-8410-435d-b29c-ffe94ff39b95)

I suggest:
1. Add `has-background-success-light` class for `<tr>`
2. Your button will have `is-info` class depends on `isSelected`
3. You can remove `minusStyle` because you can get same result without it